### PR TITLE
Add ACS, rent and property taxes and 3-year CPS

### LINKED ISSUE #31: Migrate ACS from policyengine-us
https://github.com/PolicyEngine/policyengine-us/tree/master/policyengine_us/data/datasets/acs

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #31

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Have you tried running it?

File: acs.py
Comment (nikhilwoodruff) [inline acs.py:1]: Can you remove this from the repo and add a test generating the dataset?

File: policyengine_us_data/datasets/acs/acs.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        acs["spm_unit_net_income_reported"] = spm_unit.SPM_RESOURCES
+        acs["spm_unit_spm_threshold"] = spm_unit.SPM_POVTHRESHOLD
+
+    @staticmethod
+    def add_household_variables(acs: h5py.File, household: DataFrame) -> None:
+        acs["household_vehicles_owned"] = household.VEH
+        acs["state_fips"] = acs["household_state_fips"] = household.ST.astype(int)
+
+
+class ACS_2022(ACS):
```
Comment (nikhilwoodruff) [inline policyengine_us_data/datasets/acs/acs.py:87]: LGTM, and just 2022 is fine!

File: acs.py
Comment (PavelMakarchuk) [reply acs.py:1]: I have create a test to see if the ACS is generated but nothing else - no loading or checking tables etc

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you add `acs/README.md` linking to the codebook?

Also if easy, could you add rent and property tax as variables?

**Comment by PavelMakarchuk** [comment]
> Could you add `acs/README.md` linking to the codebook?
> 
> Also if easy, could you add rent and property tax as variables?

Added a readme with documentation

We already include 
GRNTP -> Gross Rent
RNTP -> Monthly rent

I don't see variables for property taxes in the [dictionary](https://www2.census.gov/programs-surveys/acs/tech_docs/pums/data_dict/PUMS_Data_Dictionary_2022.pdf)

**Comment by MaxGhenis** [comment]
image

**Comment by PavelMakarchuk** [comment]
> image

No idea how i missed that - added

**Review by MaxGhenis — APPROVED** [review]

File: policyengine_us_data/datasets/acs/acs.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        acs["state_fips"] = acs["household_state_fips"] = household.ST.astype(
+            int
+        )
+
+
+class ACS_2022(ACS):
+    name = "acs_2022"
+    label = "ACS 2022"
+    time_period = 2022
+    file_path = STORAGE_FOLDER / "acs_2022.h5"
```
Comment (nikhilwoodruff) [inline policyengine_us_data/datasets/acs/acs.py:100]: Sorry could you just add a `url` attribute (see e.g. the ECPS, doesn't have to already exist)

File: policyengine_us_data/datasets/acs/acs.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import os
+
+
+class ACS(Dataset):
+    name = "acs"
+    label = "ACS"
+    data_format = Dataset.ARRAYS
+    time_period = None
+
+    def __init__(self):
```
Comment (nikhilwoodruff) [inline policyengine_us_data/datasets/acs/acs.py:17]: I don't think this syntax matches the other Raw [dataset] files.

**Comment by nikhilwoodruff** [comment]
@MaxGhenis a few things (thanks @PavelMakarchuk; taking over the baton for this PR) you might want to take a look at (but not essential or blocking).

* I've moved from our implementation of QRFs in survey-enhance to a more lightweight wrapper over [quantile-forest](https://zillow.github.io/quantile-forest/index.html) which seems faster and better-written than mine. We still have the mean quantile ability. The new implementation wrapper is in `utils/qrf.py`.
* Seems like we shouldn't calibrate real estate taxes from the SOI? But I can add calibration targets for total rent and property taxes, any ideas on the definitive source to use? They match the ACS totals after imputation.

**Comment by MaxGhenis** [comment]
Correct we should not calibrate real estate taxes from SOI. See #29 for other options.

**Comment by nikhilwoodruff** [comment]
SNAP is looking really nice with the rent imputation: after applying [USDA's 82% take-up rate](https://www.fns.usda.gov/usamap), we get 103bn vs 107bn in 2023.

**Comment by nikhilwoodruff** [comment]
Also tied in #67 here.

**Comment by nikhilwoodruff** [comment]
Getting memory breaches but I'm going to merge since it ran locally.

**Pull request merged**